Dockerfile

```diff
@@ -35,14 +33,14 @@ RUN apt-get update -y && \
         imagemagick \
         krb5-config \
         krb5-user \
-        libauthen-krb5-perl \
+        libauthen-krb5-simple-perl \
         libkrb5-dev \
         libssl-dev \
         make \
         pkg-config \
-        python3-dev \
+        python3.12-dev \
         python3-pip \
-        python3.8 \
+        python3.12 \
```

Comment:
BTW the Snakemake py312 f-string related fix in https://github.com/reanahub/reana-commons/pull/465 now requires `git` executable to be present in the image in order to install our Snakemake patches, so you can proactively add `git` to the list of requirements here.

Reply:
This is the diff of the changes I was using:
```diff
diff --git a/Dockerfile b/Dockerfile
index df7ca56..694f258 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -27,6 +27,7 @@ RUN apt-get update -y && \
         curl \
         g++ \
         gcc \
+        git \
         gnupg2 \
         graphviz \
         graphviz-dev \
@@ -64,6 +65,7 @@ RUN apt-get update -y && \
         cmake \
         g++ \
         gcc \
+        git \
         graphviz-dev \
         libssl-dev \
         make \
@@ -91,13 +93,22 @@ RUN if [ "${DEBUG}" -gt 0 ]; then pip install --no-cache-dir -e ".[debug,xrootd]
 
 # Are we building with locally-checked-out shared modules?
 # hadolint ignore=DL3013
-RUN if test -e modules/reana-commons; then \
+RUN apt-get update -y && \
+    apt-get install -y --no-install-recommends \
+      git && \
+    if test -e modules/reana-commons; then \
       if [ "${DEBUG}" -gt 0 ]; then \
         pip install --no-cache-dir -e "modules/reana-commons[snakemake_reports]" --upgrade; \
       else \
         pip install --no-cache-dir "modules/reana-commons[snakemake_reports]" --upgrade; \
       fi \
-    fi
+    fi && \
+    apt-get remove -y \
+        git && \
+    apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
```